.github/workflows/validate.yml

```diff
@@ -0,0 +1,101 @@
+name: Test website
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    name: Build Jekyll site for testing
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Build the site in the Jekyll/builder container
+      run: |
+        docker run \
+        -v ${{ github.workspace }}:/srv/jekyll -v ${{ github.workspace }}/_site:/srv/jekyll/_site \
```

Comment:
We have `nox` usable with this repository now, so we could just re-use that here via a verb like `nox -s build`. That might reduce some redundancy.

https://github.com/jupyter/jupyter.github.io/blob/10cdc9296f64b5607b4f7e5ab46c21d31a25e7c9/noxfile.py#L15-L18

We'd need to update that so that there was a build verb that didn't also do `liveserve` but that would be easy